File: app/src/main/java/com/example/android/pingpongscorekeeper/adapters/MatchSetsAdapter.java
package com.example.android.pingpongscorekeeper.adapters;

import android.content.Context;
import android.database.Cursor;
import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CursorAdapter;
import android.widget.TextView;

import com.example.android.pingpongscorekeeper.R;

import static com.example.android.pingpongscorekeeper.data.PingPongContract.Set.PLAYER_ONE_SCORE;
import static com.example.android.pingpongscorekeeper.data.PingPongContract.Set.PLAYER_TWO_SCORE;
import static com.example.android.pingpongscorekeeper.data.PingPongContract.Set.SET_NUMBER;

public class MatchSetsAdapter extends CursorAdapter {

    public MatchSetsAdapter(Context context, Cursor c) {
        super(context, c, 0);
    }

    @Override
    public View newView(Context context, Cursor cursor, ViewGroup parent) {
        return LayoutInflater.from(context).inflate( R.layout.details_item, parent, false);
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
        final TextView setNumberView = view.findViewById(R.id.set_n);
        final TextView playerOneScoreView = view.findViewById(R.id.player_one_num);
        final TextView playerTwoScoreView = view.findViewById(R.id.player_two_num);

        final int playerOneSetScoreCol = cursor.getColumnIndex(PLAYER_ONE_SCORE);
        final int playerTwoSetScoreCol = cursor.getColumnIndex(PLAYER_TWO_SCORE);
        final int setNumberCol = cursor.getColumnIndex(SET_NUMBER);

        final int playerOneScore = cursor.getInt(playerOneSetScoreCol);
        final int playerTwoScore = cursor.getInt(playerTwoSetScoreCol);
        final int setNumber = cursor.getInt(setNumberCol);

        setNumberView.setText("SET " + setNumber);
        playerOneScoreView.setText(playerOneScore + "");
        playerTwoScoreView.setText(playerTwoScore + "");

        playerOneScoreView.setTypeface(
                Typeface.create(playerOneScoreView.getTypeface(), Typeface.NORMAL), Typeface.NORMAL
        );
        playerTwoScoreView.setTypeface(
                Typeface.create(playerTwoScoreView.getTypeface(), Typeface.NORMAL), Typeface.NORMAL
        );

        if(playerOneScore > playerTwoScore) {
            playerOneScoreView.setTypeface(playerOneScoreView.getTypeface(), Typeface.BOLD);
        }
        else {
            playerTwoScoreView.setTypeface(playerTwoScoreView.getTypeface(), Typeface.BOLD);
        }


    }
}

File: app/src/main/java/com/example/android/pingpongscorekeeper/adapters/MatchSetsAdapter.kt
package com.example.android.pingpongscorekeeper.adapters

import android.content.Context
import android.database.Cursor
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CursorAdapter
import android.widget.TextView
import com.example.android.pingpongscorekeeper.R
import com.example.android.pingpongscorekeeper.data.PingPongContract.Set.*

class MatchSetsAdapter(context: Context, c: Cursor?)  : CursorAdapter(context, c, 0)  {

    override fun newView(context: Context, cursor: Cursor, parent: ViewGroup): View {
        return LayoutInflater.from(context).inflate(R.layout.details_item, parent, false)
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
        val setNumberView = view.findViewById<TextView>(R.id.set_n)
        val playerOneScoreView = view.findViewById<TextView>(R.id.player_one_num)
        val playerTwoScoreView = view.findViewById<TextView>(R.id.player_two_num)

        val playerOneSetScoreCol = cursor.getColumnIndex(PLAYER_ONE_SCORE)
        val playerTwoSetScoreCol = cursor.getColumnIndex(PLAYER_TWO_SCORE)
        val setNumberCol = cursor.getColumnIndex(SET_NUMBER)

        val playerOneScore = cursor.getInt(playerOneSetScoreCol)
        val playerTwoScore = cursor.getInt(playerTwoSetScoreCol)
        val setNumber = cursor.getInt(setNumberCol)

        setNumberView.text = "SET $setNumber"
        playerOneScoreView.text = playerOneScore.toString()
        playerTwoScoreView.text = playerTwoScore.toString()

        playerOneScoreView.setTypeface(
                Typeface.create(playerOneScoreView.typeface, Typeface.NORMAL), Typeface.NORMAL
        )
        playerTwoScoreView.setTypeface(
                Typeface.create(playerTwoScoreView.typeface, Typeface.NORMAL), Typeface.NORMAL
        )

        if (playerOneScore > playerTwoScore) {
            playerOneScoreView.setTypeface(playerOneScoreView.typeface, Typeface.BOLD)
        } else {
            playerTwoScoreView.setTypeface(playerTwoScoreView.typeface, Typeface.BOLD)
        }
    }
}
